### DISCUSSION TITLE
Default setting for remembering / forgetting passwords?

### DISCUSSION THREAD

**Discussion by 097115** [lxqt/lxqt#2465] [Q&A]
When, for example, an encrypted LUKS volume is inserted and the Mount window with a password prompt is popped up, can I somehow set the default setting to **Forget password immediately** instead of **Remember password for this session**?

![](https://i.imgur.com/zxGAa2t.png)

It's really an extremely annoying additional click every time :)

(And by the way, "this session" means until the next logout, is this correct?)

**Comment by tsujan** [comment]
I agree that it's annoying. Will look into it.

**Reply by tsujan** [reply]
> And by the way, "this session" means until the next logout, is this correct?

Not necessarily, if I remember correctly. I use `gnome-keyring-daemon`; its "session" seems to be defined in another way. That's out of our control.

**Reply by stefonarch** [reply]
It's at reboot here with "remember for this session". So the option's title is somehow misleading, don't know if there are other options than using gnome-keyring, never got friendly with kwallet and if it's behavior is identical.

**Reply by tsujan** [reply]
I'm sure we discussed it elsewhere. Since we can't know the session length of a password storing app, I think "Remember password for this session" is our best option.

**Reply by tsujan** [reply]
I'll add a short tooltip to the radio button in the upcoming PR: "Session length is determined by password manager"

> It's at reboot here with "remember for this session".

I may have encountered a time dependent behavior too but don't remember where or how.

**Reply by stefonarch** [reply]
As it stores also  the access data for element/matrix on startup and the never again while I have quite long uptimes (also a week) I think gnome-keyring's session is "until reboot".

**Reply by tsujan** [reply]
Yes, it may depend on whether it stores something at startup. Perhaps what I saw happened years ago, when I started to use gnome-keyring in LXQt. In that time I had Connman+CMST; now I use Network Manager.

**Comment by tsujan** [comment]
Since the mount dialog is deep inside `libfm-qt`, and so, `pcmanfm-qt` has no direct access to it, we need  a separate config file `~/.config/lxqt/mountdialog`. Then, not only how the password is remembered but also anonymity could be saved if relevant (although the latter is usually disabled). I think the required changes are elementary and backward compatible; will make a PR...

**Comment by tsujan** [comment]
The PR is here: https://github.com/lxqt/libfm-qt/pull/917
